### ISSUE TITLE
Scenic requires Python>3.8 but that causes issue with CARLA (requires v3.7)

### ISSUE THREAD

**Issue by AntiLibrary5** [BerkeleyLearnVerify/Scenic#24]
CARLA version == 0.9.11
Python version == 3.8.8

Hi, I have been trying to use the scenic API for CARLA. But having followed the instructions at: 
https://scenic-lang.readthedocs.io/en/latest/simulators.html#simulators
I run into error which of python version conflict.

Previously I have tried using Python 3.8 for CARLA but it didn't work, but worked only with v3.7. But now scenic's hard dependence on v3.8 seems problematic. Is there a workaround to this problem?

I have been using CARLA with a conda venv with Python 3.7 and it works fine.
For scenic, I created a new venev with Python 3.8 and 
>> pip install scenic 
>> easy_install /PATH_TO_CARLA_FOLDER/PythonAPI/carla/dist/carla-0.9.11-py3.7-win-amd64.egg

Upon running: >> scenic [file-name]
I get the following error:

import carla as _carla
File "c:\users\vaibh\anaconda3\envs\venv_scenic\lib\site-packages\carla-0.9.11-py3.7-win-amd64.egg\carla\__init__.py", line 8, in <module>
    from .libcarla import *
  File "c:\users\vaibh\anaconda3\envs\venv_scenic\lib\site-packages\carla-0.9.11-py3.7-win-amd64.egg\carla\libcarla.py", line 7, in <module>
    __bootstrap__()
  File "c:\users\vaibh\anaconda3\envs\venv_scenic\lib\site-packages\carla-0.9.11-py3.7-win-amd64.egg\carla\libcarla.py", line 6, in __bootstrap__
    imp.load_dynamic(__name__,__file__)
  File "c:\users\vaibh\anaconda3\envs\venv_scenic\lib\imp.py", line 342, in load_dynamic
    return _load(spec)
ImportError: Module use of python37.dll conflicts with this version of Python.

**Comment by oslundstrom** [comment]
You can solve this by compiling Carlas PythonAPI for 3.8.

**Comment by AntiLibrary5** [comment]
Hi,
Thank you for your response. The problem is building carla from source is non-trivial (mainly because of the storage requirement of ~165gb) and the packaged installation doesn't come with an egg for Python 3.8 yet. But I found the egg file for Python 3.8 and I was able to run Scenic.

For anyone running into similar issue of using the packaged installation of CARLA 0.9.11 that comes with the egg file for Python 3.7 and scenic (which needs 3.8), download the egg file from:
https://github.com/carla-simulator/carla/issues/3917#issuecomment-786568058

and then in your venv:
>>easy_install [CARLA-path]/PythonAPI/carla/dist/carla-0.9.11-py3.8-win-amd64.egg
>>scenic Scenic-master\examples\carla\Carla_Challenge\carlaChallenge1.scenic --simulate --model scenic.simulators.carla.model
